src/Merchants.php

```diff
@@ -142,15 +142,36 @@ public static function update_or_create_merchant() {
 			'merchant_name'    => $merchant_name,
 		);
 
-		// The response only contains the merchant id.
-		$response = API\Base::update_or_create_merchant( $args );
+		$cache_key = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '_request_' . md5( wp_json_encode( $args ) );
+		$cache     = get_transient( $cache_key );
+
+		if ( ! $cache ) {
```

Comment:
I think that this is wrong. If transient expires then `get_transient` will return `false` and we will get inside the `if` resulting in returned `false` value. We should use `get_transient` API directly and check for `false` explicitly:
```suggestion
		if ( false !== $cache ) {
```

Reply:
Of course :man_facepalming: 
Yes, that was the idea. Nice catch.